boards/arm/efr32bg_sltb010a/board.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021 Sateesh Kotapati
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+
+#ifdef CONFIG_SOC_GECKO_DEV_INIT
+#include "em_cmu.h"
+#endif
+
+LOG_MODULE_REGISTER(efr32bg_sltb010a, CONFIG_BOARD_EFR32BG22_LOG_LEVEL);
+
+static int efr32bg_sltb010a_init_clocks(void);
+
+static int efr32bg_sltb010a_init(const struct device *dev)
+{
+	int ret;
+
+#ifdef CONFIG_SOC_GECKO_DEV_INIT
+	efr32bg_sltb010a_init_clocks();
+#endif
+	static struct gpio_dt_spec wake_up_gpio_dev =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(wake_up_trigger), gpios);
+
+	ARG_UNUSED(dev);
+
+	if (!device_is_ready(wake_up_gpio_dev.port)) {
+		LOG_ERR("Wake-up GPIO device was not found!\n");
+		return -ENODEV;
+	}
+	ret = gpio_pin_configure_dt(&wake_up_gpio_dev, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0)
+		return ret;
+
+	return 0;
+}
+
+#ifdef CONFIG_SOC_GECKO_DEV_INIT
+int efr32bg_sltb010a_init_clocks(void)
+{
+	CMU_ClockSelectSet(cmuClock_SYSCLK, cmuSelect_HFRCODPLL);
+#if defined(_CMU_EM01GRPACLKCTRL_MASK)
+	CMU_ClockSelectSet(cmuClock_EM01GRPACLK, cmuSelect_HFRCODPLL);
+#endif
+#if defined(_CMU_EM01GRPBCLKCTRL_MASK)
+	CMU_ClockSelectSet(cmuClock_EM01GRPBCLK, cmuSelect_HFRCODPLL);
+#endif
+	CMU_ClockSelectSet(cmuClock_EM23GRPACLK, cmuSelect_LFRCO);
+	CMU_ClockSelectSet(cmuClock_EM4GRPACLK, cmuSelect_LFRCO);
+#if defined(RTCC_PRESENT)
+	CMU_ClockSelectSet(cmuClock_RTCC, cmuSelect_LFRCO);
+#endif
+	CMU_ClockSelectSet(cmuClock_WDOG0, cmuSelect_LFRCO);
+
+	return 0;
+}
+#endif
```

Comment:
while clock configuration is board dependent, we should not be handling this on a board basis IMO. Boards should define the clock configuration in Devicetree, and get it applied using generic code (see e.g. STM32). Anyway, clock control in Zephyr is already a mess, so I guess we can make an exception here as soon as there is a compromise to fix this later.

Reply:
It looks like other exx32-based boards (e.g. efm32gg_stk3701a) have their clocks configured by the board init code as well. Let's land this PR first, and continue the clk cleanup in the following PRs, as we will need to modify more than just this board (efr32bg_sltb010a).